pkg/strategy/production.go

```diff
@@ -105,8 +105,8 @@ func newProductionStrategy(jaeger *v1.Jaeger, es *storage.ElasticsearchDeploymen
 			c.secrets = es.ExtractSecrets()
 			c.elasticsearches = append(c.elasticsearches, *es.Elasticsearch())
 
-			es.InjectStorageConfiguration(&queryDep.Spec.Template.Spec)
-			es.InjectStorageConfiguration(&cDep.Spec.Template.Spec)
+			es.InjectStorageConfiguration(&queryDep.Spec.Template.Spec, !query.TokenPropagation())
```

Comment:
Why is `TokenPropagation` configuration being used to determine is `tls` is enabled? Surely they are separate concerns - i.e. `tls` may be enabled regardless of whether the token is propagated or not?

Reply:
Because if TLS is enabled (tls authentication), the token will be ignored by ES.

as far as I know, this is not disabling TLS communication, but authentication

Reply:
Unless I am misunderstanding something, the `tls` is protecting the transport/communications between the collector and ES, regardless of the originating user - whereas the token is specific to the user making the request (used for authorization) - so they are two different things?

Reply:
Not exactly, the `tls` flag  enable/disable ssl client authentication. if you disable `tls` but specify a CA, the communication will be protected .

You can see the behaviour here:

https://github.com/jaegertracing/jaeger/blob/master/pkg/es/config/config.go#L294
https://github.com/jaegertracing/jaeger/blob/master/pkg/es/config/config.go#L340

Said that, I think the flag name is a little bit confusing.

Reply:
@rubenvp8510 could you please document this in `InjectStorageConfiguration`?